packages/sessions-sdk-ts/src/index.ts

```diff
@@ -857,5 +857,15 @@ export const verifyLogInToken = async (
   );
   if (!isValid) return;
 
-  return getSessionAccount(connection, new PublicKey(payload.sessionPublicKey));
+  const sessionAccount = await getSessionAccount(
+    connection,
+    new PublicKey(payload.sessionPublicKey),
+  );
+  if (!sessionAccount) return;
+
+  if (sessionAccount.expiration.getTime() < Date.now()) {
+    throw new Error("The session associated with this login token has expired");
+  }
+
+  return sessionAccount;
```

Comment:
This can be a bit more concise:

```suggestion
  if (sessionAccount?.expiration.getTime() < Date.now()) {
    throw new Error("The session associated with this login token has expired");
  } else {
    return sessionAccount;
  }
```

I personally dislike early returns and try to avoid them, but of course you could leave out the `else` if you wanted.

Reply:
mmmm I since this function is using many early returns already I'm going to keep it like this.
Also when reading your suggestion i had to think a little bit on what happens if sessionAccount is undefined, which makes me think it reads worse if you're not deep into typescript.